Contributing doc

myronmarston opened the pull request:
Start of rspec/rspec-mocks#986.  I'm kinda basing this off of bundler's files.  More to come.

JonRowe commented:
Looks good to me so far!

myronmarston commented:
OK, I think this is ready for review.  @lizrush, as the one who originally requested this (in rspec/rspec-mocks#986), can you take a look and let us know if this is the kind of thing you were looking for?  Are there any general gotchas you ran into not covered here?

You can see how these files look when rendered:

* https://github.com/rspec/rspec-core/blob/update-common-markdown-files-2015-07-24-for-master/CONTRIBUTING.md
* https://github.com/rspec/rspec-core/blob/update-common-markdown-files-2015-07-24-for-master/DEVELOPMENT.md

myronmarston commented on common_markdown_files/CONTRIBUTING.md.erb:
```diff
@@ -0,0 +1,20 @@
+# Contributing
+
+RSpec is a community-driven project that has benefited from improvements from over over *500* contributors.
+We welcome contributions from *everyone*. While contributing, please follow the project [code of conduct](../code_of_conduct.md), so that everyone can be included.
```
Hmm, this code of conduct link doesn't appear to be working (at least not from https://github.com/rspec/rspec-core/blob/update-common-markdown-files-2015-07-24-for-master/CONTRIBUTING.md).

Any one got an idea for how to fix this?  There's also the DEVELOPMENT file link below with the same problem.

xaviershay commented on common_markdown_files/DEVELOPMENT.md.erb:
```diff
@@ -0,0 +1,246 @@
+
+## Specs
+
+RSpec dogfoods itself. It's primary defense against regressions is its spec suite. Run with:
```
no apostrophe

xaviershay commented on common_markdown_files/DEVELOPMENT.md.erb:
```diff
@@ -0,0 +1,246 @@
+
+## Gotcha: Version mismatch from sibling repos
+
+The [../Gemfile] is designed to be flexible and support using
```
missing link

xaviershay commented on common_markdown_files/DEVELOPMENT.md.erb:
```diff
@@ -0,0 +1,246 @@
+[rspec-dev](https://github.com/rspec/rspec-dev#README) or you can
+set this project up individually.
+
+## Using rspec-dev
```
I'd bury this section lower. As you mention in the text, it's not relevant to most people.

xaviershay commented:
Reading @lizrush's original request, we haven't covered everything on her list yet. (i.e. squash commits?) There's also a lot of material here, which could be intimidating. What do you think about a "Pull Request Checklist" somewhere near the top. (Maybe linking to relevant sections for more detail?)

Something like:

* Documentation reflects changes and renders correctly.
* Tests are passing.
* No new Ruby warnings.
* Rubocop passes.
* Commits are squashed.
* No changelog entry.

xaviershay commented on common_markdown_files/DEVELOPMENT.md.erb:
```diff
@@ -0,0 +1,246 @@
+to measure and enforce test coverage. If the coverage falls below a
+project-specific threshold, the build will fail.
+
+## Cukes
```
Need to describe what is appropriate for cukes. Maybe:

* Cukes are our documentation. Only user facing examples that help demonstrate usage. Anything else should be in specs.
* Duplication between YARD and cukes is ok.

myronmarston commented:
Thanks, @xaviershay.  I covered some of your feedback but there's more to do (which I'll have to do another time -- time to put the kids to bed!).

myronmarston commented:
OK, @xaviershay, I think I addressed all your feedback.  Let me know what you think of the new organization.  You can see rendered examples on the [rspec-core PR](https://github.com/rspec/rspec-core/pull/2043/files).

xaviershay commented:
Fantastic, thanks. Ship it!

myronmarston commented:
Thanks, @xaviershay.  Merged in all repos
